dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package downloader
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"time"
+
+	"github.com/dragonflyoss/Dragonfly/dfget/config"
+	"github.com/dragonflyoss/Dragonfly/dfget/core/helper"
+	"github.com/dragonflyoss/Dragonfly/dfget/util"
+)
+
+// ClientWriter writes a file for uploading and a target file.
+type ClientWriter struct {
```

Comment:
We need to add more description for every fields in struct definition.

Reply:
Maybe a a separate pull request. This pull request is only to reorganize the files.

Reply:
Has been recorded in the issue #347.